crypto/rsa/rsa_ameth.c

```diff
@@ -118,7 +118,18 @@ static int rsa_pub_decode(EVP_PKEY *pkey, X509_PUBKEY *pubkey)
 
 static int rsa_pub_cmp(const EVP_PKEY *a, const EVP_PKEY *b)
 {
-    if (BN_cmp(b->pkey.rsa->n, a->pkey.rsa->n) != 0
+    /*
+     * Don't check the public/private key, this is mostly for smart
+     * cards.
+     */
+    if (((a->type == EVP_PKEY_RSA)
+            && (RSA_flags(a->pkey.rsa) & RSA_METHOD_FLAG_NO_CHECK))
+            ||
+        ((b->type == EVP_PKEY_RSA)
+            && (RSA_flags(b->pkey.rsa) & RSA_METHOD_FLAG_NO_CHECK))
+       )
+        ; /* no-op */
```

Comment:
I would prefer simply to return here, and drop the "else" from the beginning of the line below.